File: CS/CustomSkinBarManagerPaintStyle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Painters;
using DevExpress.XtraBars.Styles;
using DevExpress.XtraBars.Controls;
using DevExpress.XtraBars.ViewInfo;

namespace WindowsFormsApplication242 {
    public class CustomSkinBarManagerPaintStyle : SkinBarManagerPaintStyle {
        public CustomSkinBarManagerPaintStyle(BarManagerPaintStyleCollection col)
            : base(col) {
        }
        protected override void CreatePainters() {
            base.CreatePainters();
            this.fPrimitivesPainter = new MySkinPrimitivesPainter(this);
        }
        public override string Name { get { return "Custom"; } }
        protected override void RegisterItemInfo() {
            base.RegisterItemInfo();
            ItemInfoCollection.Add(new BarItemInfo("CustomBarCheckItem", "CustomCheck", 9, typeof(CustomBarCheckItem), typeof(CustomBarCheckItemLink), typeof(MyBarCheckButtonLinkViewInfo), new BarCheckButtonLinkPainter(this), true, true));
        }
        protected override void RegisterBarInfo() {
            BarInfoCollection.Add(new BarControlInfo("BarDockControl", typeof(BarDockControl), typeof(DockControlOffice2003ViewInfo), new DockControlOffice2003Painter(this)));
            BarInfoCollection.Add(new BarControlInfo("BarControl", typeof(CustomControl), typeof(BarControlViewInfo), new SkinBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("DockedBarControl", typeof(DockedBarControl), typeof(SkinDockedBarControlViewInfo), new SkinBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("SubMenuControl", typeof(SubMenuBarControl), typeof(MySubMenuBarControlViewInfo), new SkinBarSubMenuPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("PopupMenuControl", typeof(PopupMenuBarControl), typeof(PopupMenuBarControlViewInfo), new SkinBarSubMenuPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("PopupContainerControl", typeof(PopupContainerBarControl), typeof(PopupContainerControlViewInfo), new SkinPopupControlContainerBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("QuickCustomizationBarControl", typeof(QuickCustomizationBarControl), typeof(QuickCustomizationBarControlViewInfo), new SkinBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("FloatingBarControl", typeof(FloatingBarControl), typeof(SkinFloatingBarControlViewInfo), new SkinFloatingBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("TitleBarControl", typeof(DevExpress.XtraBars.Objects.TitleBarControl), typeof(DevExpress.XtraBars.Objects.TitleBarControlViewInfo), new SkinTitleBarPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("ControlForm", typeof(DevExpress.XtraBars.Forms.ControlForm), typeof(ControlFormViewInfo), new SkinControlFormPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("SubMenuControlForm", typeof(DevExpress.XtraBars.Forms.SubMenuControlForm), typeof(ControlFormViewInfo), new SkinControlFormPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("PopupContainerControlForm", typeof(DevExpress.XtraBars.Forms.PopupContainerForm), typeof(PopupContainerFormViewInfo), new SkinControlFormPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("FloatingBarControlForm", typeof(DevExpress.XtraBars.Forms.FloatingBarControlForm), typeof(SkinFloatingBarControlFormViewInfo), new SkinFloatingBarControlFormPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("GalleryDropDownBarControl", typeof(GalleryDropDownBarControl), typeof(GalleryDropDownControlViewInfo), new GalleyDropDownBarControlPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("AppMenuBarControl", typeof(AppMenuBarControl), typeof(AppMenuControlViewInfo), new BarSubMenuPainter(this)));
            BarInfoCollection.Add(new BarControlInfo("AppMenuForm", typeof(DevExpress.XtraBars.Forms.AppMenuForm), typeof(AppMenuFormViewInfo), new SkinAppMenuFormPainter(this)));
        }
        int FindBarInfo(Type itemType) {
            for(int i = 0; i < BarInfoCollection.Count; i++) {
                if(BarInfoCollection[i].ItemType == itemType) return i;
            }
            return -1;
        }
        
    }
    public class CustomBarCheckItemLink:BarCheckItemLink {
        public CustomBarCheckItemLink(BarItemLinkReadOnlyCollection ALinks, BarItem AItem, object ALinkedObject)
            : base(ALinks, AItem, ALinkedObject) {
        }
    }
    public class CustomBarCheckItem:BarCheckItem {
        public CustomBarCheckItem():base() {

        }
        public CustomBarCheckItem(BarManager manager)
            : base(manager) {
        }
        public CustomBarCheckItem(BarManager manager, bool check)
            : base(manager, check) {

        } 
    }

}

File: VB/CustomSkinBarManagerPaintStyle.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Painters
Imports DevExpress.XtraBars.Styles
Imports DevExpress.XtraBars.Controls
Imports DevExpress.XtraBars.ViewInfo

Namespace WindowsFormsApplication242
    Public Class CustomSkinBarManagerPaintStyle
        Inherits SkinBarManagerPaintStyle

        Public Sub New(ByVal col As BarManagerPaintStyleCollection)
            MyBase.New(col)
        End Sub
        Protected Overrides Sub CreatePainters()
            MyBase.CreatePainters()
            Me.fPrimitivesPainter = New MySkinPrimitivesPainter(Me)
        End Sub
        Public Overrides ReadOnly Property Name() As String
            Get
                Return "Custom"
            End Get
        End Property
        Protected Overrides Sub RegisterItemInfo()
            MyBase.RegisterItemInfo()
            ItemInfoCollection.Add(New BarItemInfo("CustomBarCheckItem", "CustomCheck", 9, GetType(CustomBarCheckItem), GetType(CustomBarCheckItemLink), GetType(MyBarCheckButtonLinkViewInfo), New BarCheckButtonLinkPainter(Me), True, True))
        End Sub
        Protected Overrides Sub RegisterBarInfo()
            BarInfoCollection.Add(New BarControlInfo("BarDockControl", GetType(BarDockControl), GetType(DockControlOffice2003ViewInfo), New DockControlOffice2003Painter(Me)))
            BarInfoCollection.Add(New BarControlInfo("BarControl", GetType(CustomControl), GetType(BarControlViewInfo), New SkinBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("DockedBarControl", GetType(DockedBarControl), GetType(SkinDockedBarControlViewInfo), New SkinBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("SubMenuControl", GetType(SubMenuBarControl), GetType(MySubMenuBarControlViewInfo), New SkinBarSubMenuPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("PopupMenuControl", GetType(PopupMenuBarControl), GetType(PopupMenuBarControlViewInfo), New SkinBarSubMenuPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("PopupContainerControl", GetType(PopupContainerBarControl), GetType(PopupContainerControlViewInfo), New SkinPopupControlContainerBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("QuickCustomizationBarControl", GetType(QuickCustomizationBarControl), GetType(QuickCustomizationBarControlViewInfo), New SkinBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("FloatingBarControl", GetType(FloatingBarControl), GetType(SkinFloatingBarControlViewInfo), New SkinFloatingBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("TitleBarControl", GetType(DevExpress.XtraBars.Objects.TitleBarControl), GetType(DevExpress.XtraBars.Objects.TitleBarControlViewInfo), New SkinTitleBarPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("ControlForm", GetType(DevExpress.XtraBars.Forms.ControlForm), GetType(ControlFormViewInfo), New SkinControlFormPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("SubMenuControlForm", GetType(DevExpress.XtraBars.Forms.SubMenuControlForm), GetType(ControlFormViewInfo), New SkinControlFormPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("PopupContainerControlForm", GetType(DevExpress.XtraBars.Forms.PopupContainerForm), GetType(PopupContainerFormViewInfo), New SkinControlFormPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("FloatingBarControlForm", GetType(DevExpress.XtraBars.Forms.FloatingBarControlForm), GetType(SkinFloatingBarControlFormViewInfo), New SkinFloatingBarControlFormPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("GalleryDropDownBarControl", GetType(GalleryDropDownBarControl), GetType(GalleryDropDownControlViewInfo), New GalleyDropDownBarControlPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("AppMenuBarControl", GetType(AppMenuBarControl), GetType(AppMenuControlViewInfo), New BarSubMenuPainter(Me)))
            BarInfoCollection.Add(New BarControlInfo("AppMenuForm", GetType(DevExpress.XtraBars.Forms.AppMenuForm), GetType(AppMenuFormViewInfo), New SkinAppMenuFormPainter(Me)))
        End Sub
        Private Function FindBarInfo(ByVal itemType As Type) As Integer
            For i As Integer = 0 To BarInfoCollection.Count - 1
                If BarInfoCollection(i).ItemType Is itemType Then
                    Return i
                End If
            Next i
            Return -1
        End Function

    End Class
    Public Class CustomBarCheckItemLink
        Inherits BarCheckItemLink

        Public Sub New(ByVal ALinks As BarItemLinkReadOnlyCollection, ByVal AItem As BarItem, ByVal ALinkedObject As Object)
            MyBase.New(ALinks, AItem, ALinkedObject)
        End Sub
    End Class
    Public Class CustomBarCheckItem
        Inherits BarCheckItem

        Public Sub New()
            MyBase.New()

        End Sub
        Public Sub New(ByVal manager As BarManager)
            MyBase.New(manager)
        End Sub
        Public Sub New(ByVal manager As BarManager, ByVal check As Boolean)
            MyBase.New(manager, check)

        End Sub
    End Class

End Namespace
